aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -22,6 +22,18 @@ static inline void check_for_unsupported_isin_dtype(const ScalarType type) {
       "Unsupported input type encountered for isin(): ", type);
 }
 
+TORCH_META_FUNC(_s_where) (const Tensor& condition, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(self.dtype() == other.dtype(), "expected scalar type ", self.dtype(), " but found ", other.dtype());
+  Tensor cond_bool = condition.scalar_type() == ScalarType::Byte ? condition.to(ScalarType::Bool) : condition;
```

Comment:
This is not very good but I'm not sure there is necessarily a better way to do this. What is not good about it is that meta functions are not supposed to do any nontrivial compute, because they may get invoked as part of, e.g., just shape propagation (meta functions).  If condition happens to be a meta tensor this will be relatively cheap, but something that is supposed to work is, for example, calling meta::_s_where(cpu_tensors...) and that will useless do some compute in this case.

What makes matters a little better is that TensorIterator::build (used everywhere here) DOES do some temporary allocations, and it's kind of hacked around by just checking if *any* input argument is meta, and then avoiding allocating temporaries if it's the case. So I'm willing to let this slide, but let's have a comment+issue about it.

cc @ysiraichi

Reply:
Sure, I'll add a comment - it's also worth noting that this byte -> bool conversion is deprecated, so the conversion will not be needed soon hopefully